src/bin/hexyl.rs

```diff
@@ -337,6 +348,25 @@ fn run() -> Result<(), AnyhowError> {
         2
     };
 
+    let group_bytes = if let Some(group_bytes) = matches
+        .get_one::<String>("group_bytes")
+        .map(|s| {
+            s.parse::<NonZeroU8>().map(u8::from).context(anyhow!(
+                "failed to parse `-g` arg {:?} as unsigned nonzero integer",
+                s
+            ))
+        })
+        .transpose()?
+    {
+        if (group_bytes <= 8) && ((group_bytes & (group_bytes - 1)) == 0) {
+            group_bytes
+        } else {
+            1
+        }
```

Comment:
I think I would prefer a hard error (in case someone uses an illegal value for `group_bytes`), rather than a silent fallback to 1. Or why did you decide to implement it this waay?